discovery/ionos/server.go

```diff
@@ -0,0 +1,170 @@
+	serverNameLabel             = serverLabelPrefix + "name"
+	serverNICIDLabel            = serverLabelPrefix + "nic_id"
+	serverPrimaryIPLabel        = serverLabelPrefix + "primary_ip"
+	serverPrimaryLANIDLabel     = serverLabelPrefix + "primary_lan_id"
+	serverPrimaryNICIDLabel     = serverLabelPrefix + "primary_nic_id"
+	serverStateLabel            = serverLabelPrefix + "state"
+	serverTypeLabel             = serverLabelPrefix + "type"
+)
+
+type serverDiscovery struct {
+	*refresh.Discovery
+	client       *ionoscloud.APIClient
+	port         int
+	datacenterID string
+}
+
+func newServerDiscovery(conf *SDConfig, logger log.Logger) (*serverDiscovery, error) {
+	d := &serverDiscovery{
+		port:         conf.Port,
+		datacenterID: conf.DatacenterID,
+	}
+
+	rt, err := config.NewRoundTripperFromConfig(conf.HTTPClientConfig, "ionos_sd")
+	if err != nil {
+		return nil, err
+	}
+
+	// username, password and token are set via http client config
+	cfg := ionoscloud.NewConfiguration("", "", "", conf.ionosEndpoint)
+	cfg.HTTPClient = &http.Client{
+		Transport: rt,
+		Timeout:   time.Duration(conf.RefreshInterval),
+	}
+	cfg.UserAgent = fmt.Sprintf("Prometheus/%s", version.Version)
+
+	d.client = ionoscloud.NewAPIClient(cfg)
+
+	return d, nil
+}
+
+func (d *serverDiscovery) refresh(ctx context.Context) ([]*targetgroup.Group, error) {
+	api := d.client.ServersApi
+
+	servers, _, err := api.DatacentersServersGet(ctx, d.datacenterID).
+		Depth(3).
+		Execute()
+	if err != nil {
+		return nil, err
+	}
+
+	var targets []model.LabelSet
+	for _, server := range *servers.Items {
+		labels := model.LabelSet{
+			serverAvailabilityZoneLabel: model.LabelValue(*server.Properties.AvailabilityZone),
+			serverCPUFamilyLabel:        model.LabelValue(*server.Properties.CpuFamily),
+			serverDatacenterIDLabel:     model.LabelValue(d.datacenterID),
+			serverIDLabel:               model.LabelValue(*server.Id),
+			serverLifecycleLabel:        model.LabelValue(*server.Metadata.State),
+			serverNameLabel:             model.LabelValue(*server.Properties.Name),
+			serverStateLabel:            model.LabelValue(*server.Properties.VmState),
+			serverTypeLabel:             model.LabelValue(*server.Properties.Type),
+		}
+
+		if server.Properties.BootCdrom != nil {
+			labels[serverBootCDROMIDLabel] = model.LabelValue(*server.Properties.BootCdrom.Id)
+		}
+
+		if server.Properties.BootVolume != nil {
+			labels[serverBootVolumeIDLabel] = model.LabelValue(*server.Properties.BootVolume.Id)
+		}
+
+		if server.Entities != nil && server.Entities.Nics != nil {
+			var (
+				nicIDs []string
+				ips    []string
+				lans   []string
+			)
+
+			for _, nic := range *server.Entities.Nics.Items {
+				nicIDs = append(nicIDs, *nic.Id)
+				ips = append(ips, *nic.Properties.Ips...)
+				lans = append(lans, strconv.Itoa(int(*nic.Properties.Lan)))
+			}
+
+			if len(nicIDs) > 0 {
+				labels[serverPrimaryNICIDLabel] = model.LabelValue(nicIDs[0])
```

Comment:
It sounds limiting to only have one Ip. can we have the other IP's in other labels?

Reply:
Agreed and already implemented. I think you missed the comma seperated labels for ip, nic and lan :). Or am I mistaken?

Reply:
In general, this is more useful with dynamic labels `__meta_ionos_server_lan_ip_<lan_id> = <ip address>`

Reply:
Thanks for explaining. I've replaced the NIC labels with your suggested dynamic label.